### DISCUSSION TITLE
blog/deploying-ruby-on-rails-with-dokku-redis-sidekiq-arm-docker-hetzner

### DISCUSSION THREAD

**Discussion by giscus** [harrison-broadbent/railsnotes-comments#5] [RailsNotes Comments]
# blog/deploying-ruby-on-rails-with-dokku-redis-sidekiq-arm-docker-hetzner

This is the ultimate Dokku + Ruby on Rails guide — I cover running Dokku on x86 and ARM, deploying Sidekiq and Redis, dockerizing our Ruby on Rails app, and more...

https://railsnotes.xyz/blog/deploying-ruby-on-rails-with-dokku-redis-sidekiq-arm-docker-hetzner

**Comment by saint-james-fr** [comment]
Thanks for the amazing article, following along I encoutered two issues which could be mentioned : 
- `dokku proxy:ports-set <app_name> http:80:5000`this command does not run anymore on dokku CLI you have to use `dokku ports:set http:80:5000`. 
- Also I'm pretty sure i "did" have to use it without using following x86 way, container port (:5000) was not correctly configured otherwise.
- Before installing TLS certificates, don't forget to configure environment/production.rb `  config.force_ssl = false`, or it will result in a 301 status and an access error to port `Failed to connect to XXXX port 443`

**Reply by harrison-broadbent** [reply]
Thank you for the feedback! I've updated the post and mentioned your comment in the update log. Let me know if you spot anything else!

**Reply by saint-james-fr** [reply]
Hey, just was you added 
```
Note: make sure you have config.force_ssl = false in your environment/production.rb file inside your Rails app. By default it's commented out, which is fine too. Having force_ssl = true can lead to errors (Failed to connect to XXXX port 443).
```

Actually it's not exactly correct, what I meant is if you test you app before enabling TLS certificates, this should be set to false. Setting it to true once they're installed is fine. Maybe it's only my default configuration but this was set to true for me.

**Reply by harrison-broadbent** [reply]
It took me a while to get back to fix this, but I've updated it now. Thank you!

**Comment by gmcamposano** [comment]
Hi there. I have followed everything you mentioned in the post - running a basic rails app. It get to the part where is shares the url and once i click -- no services are found running there. I have been debugging for hours to see what the hell I am doing wrong and cant find it. I am using an ARM server and rails 7.1 with the default Dockerfile which is created.

**Reply by harrison-broadbent** [reply]
Hey that sounds frustrating! Have you followed the ARM section of this guide exactly? 

https://railsnotes.xyz/blog/deploying-ruby-on-rails-with-dokku-redis-sidekiq-arm-docker-hetzner#deploying-ruby-on-rails-with-dokku-to-arm-with-docker

You also need to make sure you expose the web server port using — 
```
dokku ports:set test-rails http:80:3000
```

**Reply by saint-james-fr** [reply]
Would be happy to help as well, can you provide a screenshot of what you get?

**Reply by gmcamposano** [reply]
Nailed it! What was happening? I get redirected to https with the forced ssl config. Also I forgot to use:
dokku ports:set test-rails http:80:3000. With that issue out of the way it worked. I will now follow the rest of the tutorial. Thank you guys.

**Reply by saint-james-fr** [reply]
You're welcome, enjoy Dokku! It's really the best!

**Comment by jimmypoulsen** [comment]
What an amazing guide. Literally had no obstacles following this. Worked like a charm 🔥

**Comment by yatish27** [comment]
Can you update this with Rails 8, since it comes with a default Dockerfile
